lib/Parse/ParseStmt.cpp

```diff
@@ -1917,18 +1917,23 @@ ParserResult<Stmt> Parser::parseStmtDo(LabeledStmtInfo labelInfo) {
       DoCatchStmt::create(Context, labelInfo, doLoc, body.get(), allClauses));
   }
 
-  SourceLoc whileLoc;
-
-  // If we don't see a 'while', this is just the bare 'do' scoping
-  // statement.
-  if (!consumeIf(tok::kw_while, whileLoc)) {
+  // If we dont see a 'while' or see a 'while' that starts
+  // from new line. This is just the bare `do` scoping statement.
+  if (Tok.getKind() != tok::kw_while) {
     return makeParserResult(status,
-                         new (Context) DoStmt(labelInfo, doLoc, body.get()));
+                            new (Context) DoStmt(labelInfo, doLoc, body.get()));
   }
-
-  // But if we do, advise the programmer that it's 'repeat' now.
+  SourceLoc whileLoc = Tok.getLoc();
+  if (Tok.isAtStartOfLine()) {
+    return makeParserResult(status,
+                          new (Context) DoStmt(labelInfo, doLoc, body.get()));
+  }
+    // But if we do, advise the programmer that it's 'repeat' now.
   diagnose(doLoc, diag::do_while_now_repeat_while)
-    .fixItReplace(doLoc, "repeat");
+    .fixItReplace(doLoc, "repeat")
+    .fixItInsert(whileLoc, "\n");
```

Comment:
I don't think this is right.
If the user applies these fix-its, the code will be something like:
```swift
repeat {
}
while condition
```
The newline is not needed at all.

If we want users to choose between `repeat` and bare `do`, you need to emit 3 diagnostics:
```
error: 'do-while' statement is not allowed.
note: did you mean 'repeat-file' statement?  (with fix-it to 'repeat')
note: did you mean separate 'do' and 'while' statements? (with fix-it inserting '\n')
```